core/src/main/kotlin/event/Event.kt

```diff
@@ -45,3 +51,21 @@ public interface Event : KordObject {
     @KordPreview
     public val customContext: Any?
 }
+
+/**
+ * Representation of an event that has not been documented / added to Kord yet.
+ *
+ * @property data the raw [json body][JsonElement] of this event if provided
+ * @property name the name of the event if provided
+ */
+public class UnknownEvent(
+    public val name: String?,
+    public val data: JsonElement?,
+    override val kord: Kord,
+    override val shard: Int,
+    override val customContext: Any?,
+) : Event {
+    override fun toString(): String {
+        return "UnknownEvent(kord=$kord, shard=$shard, customContext=$customContext, data=$data, name=$name)"
+    }
```

Comment:
```suggestion
    override fun toString(): String =
        "UnknownEvent(name=$name, data=$data, kord=$kord, shard=$shard, customContext=$customContext)"
```
to match the order of the constructor parameters